src/org/opendatakit/briefcase/ui/export/ExportPanel.java

```diff
@@ -16,412 +16,125 @@
 
-  private void showErrors(List<String> errors, String headerText, String title) {
-    if (!errors.isEmpty()) {
-      String message = String.format("%s\n\n%s", headerText, errors.stream().map(e -> "- " + e).collect(joining("\n")));
-      showErrorDialog(this, message, title);
-    }
-  }
-
-  /**
-   * The DatePicker default text box and calendar button don't match with the rest of the UI.
-   * This tweaks those elements to be consistent with the rest of the application.
-   */
-  private DatePicker createDatePicker() {
-    DatePicker datePicker = new DatePicker();
-    JTextField model = new JTextField();
-
-    datePicker.getComponentToggleCalendarButton().setText("Choose...");
-    datePicker.getComponentDateTextField().setBorder(model.getBorder());
-    datePicker.getComponentDateTextField().setMargin(model.getMargin());
-
-    return datePicker;
-  }
-
-  @Override
-  public void setEnabled(boolean enabled) {
-    super.setEnabled(enabled);
-    updateForms();
-
-    for (Component c : this.getComponents()) {
-      c.setEnabled(enabled);
-    }
-    if (enabled) {
-      // and then update the widgets based upon the transfer state
-      setActiveExportState(exportStateActive);
-    }
-  }
-
-  private void setActiveExportState(boolean active) {
-    if (active) {
-      // don't allow normal actions when we are transferring...
-      btnChooseExportDirectory.setEnabled(false);
-      btnPemFileChooseButton.setEnabled(false);
-      pickStartDate.setEnabled(false);
-      pickEndDate.setEnabled(false);
-      btnExport.setEnabled(false);
-      // reset the termination future so we can cancel activity
-      terminationFuture.reset();
-    } else {
-      btnChooseExportDirectory.setEnabled(true);
-      btnPemFileChooseButton.setEnabled(true);
-      pickStartDate.setEnabled(true);
-      pickEndDate.setEnabled(true);
-      btnExport.setEnabled(true);
-      // touch-up with real state...
-      updateExportButton();
-      // retain progress text (to display last export outcome)
-    }
-    // remember state...
-    exportStateActive = active;
-  }
-
-  void resetExport() {
-  }
-
-  private List<String> export() {
-    return tableModel.getSelectedForms().parallelStream()
+  private List<String> export(ExportConfiguration defaultConfiguration) {
+    form.disableUI();
+    terminationFuture.reset();
+    List<String> errors = forms.getSelectedForms().parallelStream()
         .peek(FormStatus::clearStatusHistory)
         .map(formStatus -> (BriefcaseFormDefinition) formStatus.getFormDefinition())
-        .flatMap(formDefinition -> this.export(formDefinition).stream())
+        .flatMap(formDefinition -> this.export(defaultConfiguration, formDefinition).stream())
         .collect(toList());
+    form.enableUI();
+    return errors;
   }
 
-  private List<String> export(BriefcaseFormDefinition formDefinition) {
+  private List<String> export(ExportConfiguration defaultConfiguration, BriefcaseFormDefinition formDefinition) {
```

Comment:
Should this method and the one above both have the name `export`? Perhaps either or both could have a more descriptive name. (This method is called only by the one above with the same name.)

Reply:
So, here's what I've done: I've moved the low level export method (the one that does the exporting) to `ExportAction` class to be more coherent. This `ExportPanel` shouldn't have to know so many details about exporting forms and this way we can work the design around `ExportAction`, `ExportToCSV`, etc. classes